src/test/java/com/fasterxml/jackson/databind/util/UnlimitedLookupCacheTest.java

```diff
@@ -0,0 +1,38 @@
+package com.fasterxml.jackson.databind.util;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class UnlimitedLookupCacheTest {
+    @Test
+    public void testCache() {
+        UnlimitedLookupCache<Long, String> cache = new UnlimitedLookupCache<>(4, 10);
+        assertNull(cache.get(1000L));
+        assertNull(cache.put(1000L, "Thousand"));
+        assertEquals("Thousand", cache.get(1000L));
+        assertEquals("Thousand", cache.putIfAbsent(1000L, "Míle"));
+        assertEquals("Thousand", cache.get(1000L));
+        assertEquals("Thousand", cache.put(1000L, "Míle"));
+        assertEquals("Míle", cache.get(1000L));
+        cache.clear();
+        assertNull(cache.put(1000L, "Thousand"));
+    }
+
+    @Test
+    public void testCompatibility() throws JsonProcessingException {
+        UnlimitedLookupCache<Object, JavaType> cache = new UnlimitedLookupCache<>(4, 10);
+        TypeFactory tf = TypeFactory.defaultInstance().withCache(cache);
+
+        //TODO find way to inject the `tf` instance into an ObjectMapper (via MapperBuilder?)
```

Comment:
@cowtowncoder any suggestions on best place to do this?

Reply:
Best place to... ? Inject cache?

Builder is the new mechanism, in general, but in case of `TypeFactory` I think it is reasonable that caller would need to create and configure it directly, pass instance.

Reply:
For other caches, however, need to think about it... they are held on to by `DeserializationContext` / `SerializerProvider`, and we'd ideally avoid making developers create instances of those.